tests/Fixtures/Integration/misc/PHP8_2.test

```diff
@@ -0,0 +1,179 @@
+--TEST--
+PHP 8.2 test.
+--RULESET--
+{
+    "@PhpCsFixer": true,
+    "@PHP82Migration": true
+}
+--REQUIREMENTS--
+{"php": 80200}
+--EXPECT--
+<?php
+
+// https://wiki.php.net/rfc/readonly_classes
+readonly final class Foo
+{
+    public string $c;
+    public string $prop;
+    public string $b;
+}
+
+// https://wiki.php.net/rfc/null-false-standalone-types
+class FalseNull
+{
+    public function __construct(
+        private false $falsy = false,
+        private null $nully = null
+    ) {
+    }
+
+    public function falsy(): false
+    {
+        return $this->falsy;
+    }
+
+    public function nully(): null
+    {
+        return $this->nully;
+    }
+}
+
+// https://wiki.php.net/rfc/true-type
+class TrueType
+{
+    public function __construct(
+        private true $havingABud = true
+    ) {
+    }
+
+    public function havingABud(): true
```

Comment:
I think we need `true` as function param's type too. Similarly for `false` and `null`. WDYT?

Reply:
I don't think it is possible. We already had the default type support, like `function Foo($a = true)`, but it cannot be used as type AFAIK (i.e. that would be  `function Foo(true $a)`)

Reply:
@SpacePossum From the [RFC](https://wiki.php.net/rfc/null-false-standalone-types):

> Add support for using null and false as stand-alone type declarations, wherever type declarations are currently allowed.

Since these are standalone types, they can be used everywhere, also in function params 🙂

Reply:
Are you sure?
The RFC shows it as return type hint only, so not as variable type, i.e. `function Foo(true $a)` is invalid.

 `Foo(true)` and `function Foo($a = true)` are of course allowed, but this we already support in the code base.

Reply:
I've added more fixtures where those types are used, maybe it's good enough now 🙂